pkg/api/v1alpha1/kepler_internal_types.go

```diff
@@ -53,8 +54,71 @@ type OpenShiftSpec struct {
 
 // KeplerInternalSpec defines the desired state of KeplerInternal
 type KeplerInternalSpec struct {
-	Exporter  InternalExporterSpec `json:"exporter"`
-	OpenShift OpenShiftSpec        `json:"openshift,omitempty"`
+	Exporter    InternalExporterSpec     `json:"exporter"`
+	Estimator   *InternalEstimatorSpec   `json:"estimator,omitempty"`
+	ModelServer *InternalModelServerSpec `json:"modelServer,omitempty"`
+	OpenShift   OpenShiftSpec            `json:"openshift,omitempty"`
+}
+
+// Kepler Model Server Spec
+type InternalModelServerSpec struct {
+
+	// +kubebuilder:default=true
+	Enabled bool `json:"enabled,omitempty"`
+
+	Image string `json:"image,omitempty"`
+
+	// +kubebuilder:default=""
+	URL string `json:"url,omitempty"`
+
+	// +kubebuilder:default=8100
```

Comment:
could you please add validation on min and max ?

Reply:
Done. I copied the exporter range.